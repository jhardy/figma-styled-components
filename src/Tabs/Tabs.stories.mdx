import { Tabs, Tab, TabList, TabPanels, TabPanel } from './Tabs'
import { Story, Meta } from '@storybook/addon-docs/blocks'

<Meta title='MDX|Tabs' component={Tabs} />

# Tabs

<Story name='Tabs'>
  <div className='component-box'>
    <Tabs>
      <TabList>
        <Tab>Design</Tab>
        <Tab>Prototype</Tab>
        <Tab>Code</Tab>
      </TabList>
      <TabPanels>
        <TabPanel>panel 1</TabPanel>
        <TabPanel>panel 2</TabPanel>
        <TabPanel>panel 3</TabPanel>
      </TabPanels>
    </Tabs>
  </div>
</Story>

```
import { Tabs, Tab, TabList, TabPanels, TabPanel } from 'figma-styled-components'

<Tabs>
  <TabList>
    <Tab>Design</Tab>
    <Tab>Prototype</Tab>
    <Tab>Code</Tab>
  </TabList>
  <TabPanels>
    <TabPanel>panel 1</TabPanel>
    <TabPanel>panel 2</TabPanel>
    <TabPanel>panel 3</TabPanel>
  </TabPanels>
</Tabs>
```
