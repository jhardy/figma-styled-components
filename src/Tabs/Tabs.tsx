import {
  Tab as ReachTab,
  TabList as ReachTabList,
  TabPanel as ReachPanel,
  TabPanels as ReachPanels,
  Tabs as ReachTabs
} from '@reach/tabs'
import React from 'react'
import styled from 'styled-components'
import { SectionTitle } from '../SectionTitle'

export const Tabs = styled(ReachTabs)``


export const TabList = styled(ReachTabList)`
  border-bottom: solid 1px rgba(0,0,0,0.1);
`

const StyledTab = styled(ReachTab)`
  border: none;
  background: #fff;
  outline: none;
  padding: 0 8px;
  ${SectionTitle} {
    opacity: 0.5;
  }

  &[aria-selected='true'] {
    ${SectionTitle} {
      opacity: 1;
    }
  }

  &:hover {
    ${SectionTitle} {
      opacity: 1;
    }
  }
`

export const Tab: React.FC = (props) => {
  return (
    <StyledTab>
      <SectionTitle>{props.children}</SectionTitle>
    </StyledTab>
  )
}

export const TabPanels = styled(ReachPanels)``
export const TabPanel = styled(ReachPanel)``
