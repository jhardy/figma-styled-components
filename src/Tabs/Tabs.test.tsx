import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { Tab, TabList, TabPanel, TabPanels, Tabs } from './Tabs'


test('A set of tabs', () => {
  const tree = renderer.create(<Tabs>
    <TabList>
      <Tab>Design</Tab>
      <Tab>Prototype</Tab>
      <Tab>Code</Tab>
    </TabList>
    <TabPanels>
      <TabPanel>panel 1</TabPanel>
      <TabPanel>panel 2</TabPanel>
      <TabPanel>panel 3</TabPanel>
    </TabPanels>
  </Tabs>).toJSON()
  expect(tree).toMatchSnapshot()
})
