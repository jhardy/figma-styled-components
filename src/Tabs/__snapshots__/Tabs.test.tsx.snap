// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A set of tabs 1`] = `
.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  color: rgba(0,0,0,0.8);
  background-color: #ffffff;
  font-weight: 600;
}

.c0 {
  border-bottom: solid 1px rgba(0,0,0,0.1);
}

.c1 {
  border: none;
  background: #fff;
  outline: none;
  padding: 0 8px;
}

.c1 .c2 {
  opacity: 0.5;
}

.c1[aria-selected='true'] .c2 {
  opacity: 1;
}

.c1:hover .c2 {
  opacity: 1;
}

<div
  className=""
  data-orientation="horizontal"
  data-reach-tabs=""
>
  <div
    aria-orientation="horizontal"
    className="c0"
    data-reach-tab-list=""
    onKeyDown={[Function]}
    role="tablist"
  >
    <button
      aria-controls="tabs--1--panel---1"
      aria-selected={false}
      className="c1"
      data-orientation="horizontal"
      data-reach-tab=""
      id="tabs--1--tab---1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      role="tab"
      tabIndex={-1}
      type="button"
    >
      <span
        className="sc-AxjAm c2 c3"
      >
        Design
      </span>
    </button>
    <button
      aria-controls="tabs--1--panel---1"
      aria-selected={false}
      className="c1"
      data-orientation="horizontal"
      data-reach-tab=""
      id="tabs--1--tab---1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      role="tab"
      tabIndex={-1}
      type="button"
    >
      <span
        className="sc-AxjAm c2 c3"
      >
        Prototype
      </span>
    </button>
    <button
      aria-controls="tabs--1--panel---1"
      aria-selected={false}
      className="c1"
      data-orientation="horizontal"
      data-reach-tab=""
      id="tabs--1--tab---1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      role="tab"
      tabIndex={-1}
      type="button"
    >
      <span
        className="sc-AxjAm c2 c3"
      >
        Code
      </span>
    </button>
  </div>
  <div
    className=""
    data-reach-tab-panels=""
  >
    <div
      aria-labelledby="tabs--1--tab---1"
      className=""
      data-reach-tab-panel=""
      hidden={true}
      id="tabs--1--panel---1"
      role="tabpanel"
      tabIndex={-1}
    >
      panel 1
    </div>
    <div
      aria-labelledby="tabs--1--tab---1"
      className=""
      data-reach-tab-panel=""
      hidden={true}
      id="tabs--1--panel---1"
      role="tabpanel"
      tabIndex={-1}
    >
      panel 2
    </div>
    <div
      aria-labelledby="tabs--1--tab---1"
      className=""
      data-reach-tab-panel=""
      hidden={true}
      id="tabs--1--panel---1"
      role="tabpanel"
      tabIndex={-1}
    >
      panel 3
    </div>
  </div>
</div>
`;
